doc/sphinx/language/core/variants.rst

```diff
@@ -20,7 +20,7 @@ form of the :term:`inhabitants <inhabitant>` of a variant type is done by case a
 using the `match` expression.
 
 When a constructor of a type takes an argument of that same type,
-the type becomes recursive, in which case it can be either
+the type becomes :gdef:`recursive <recursive type>`, in which case it can be either
```

Comment:
```suggestion
the type is a :gdef:`recursive type`, in which case it can be either
```

Better to show the term that appears in the glossary as is.

Reply:
you can make a followup PR, this one is merged